client/my-sites/plugins/disconnect-jetpack/disconnect-jetpack-button.jsx

```diff
@@ -3,36 +3,95 @@
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
 import Button from 'components/button';
-import DisconnectJetpackDialog from 'my-sites/plugins/disconnect-jetpack/disconnect-jetpack-dialog';
+import DisconnectJetpackDialog from 'blocks/disconnect-jetpack-dialog';
 import { recordGoogleEvent } from 'state/analytics/actions';
+import { disconnect } from 'state/jetpack/connection/actions';
+import { disconnectedSite as disconnectedSiteDeprecated } from 'lib/sites-list/actions';
+import { setAllSitesSelected } from 'state/ui/actions';
+import { getCurrentPlan } from 'state/sites/plans/selectors';
+import { getPlanClass } from 'lib/plans/constants';
+import { successNotice, errorNotice, infoNotice, removeNotice } from 'state/notices/actions';
 
 class DisconnectJetpackButton extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = { dialogVisible: false };
+	}
+
 	handleClick = ( event ) => {
 		event.preventDefault();
-		if ( this.props.isMock ) {
+		const {
+			isMock,
+			recordGoogleEvent: recordGAEvent
+		} = this.props;
+
+		if ( isMock ) {
 			return;
 		}
+		this.setState( { dialogVisible: true } );
+		recordGAEvent( 'Jetpack', 'Clicked To Open Disconnect Jetpack Dialog' );
+	};
 
-		if ( this.refs.dialog ) {
-			this.refs.dialog.getWrappedInstance().open();
-		}
+	hideDialog = () => {
+		const { recordGoogleEvent: recordGAEvent } = this.props;
+		this.setState( { dialogVisible: false } );
+		recordGAEvent( 'Jetpack', 'Clicked To Cancel Disconnect Jetpack Dialog' );
+	}
 
-		this.props.recordGoogleEvent( 'Jetpack', 'Clicked To Open Disconnect Jetpack Dialog' );
-	};
+	disconnectJetpack = () => {
+		const {
+			site,
+			translate,
+			successNotice: showSuccessNotice,
+			errorNotice: showErrorNotice,
+			infoNotice: showInfoNotice,
+			removeNotice: removeInfoNotice,
+			disconnect: disconnectSite,
+			recordGoogleEvent: recordGAEvent
+		} = this.props;
+
+		this.setState( { dialogVisible: false } );
+		recordGAEvent( 'Jetpack', 'Clicked To Confirm Disconnect Jetpack Dialog' );
+
+		const { notice } = showInfoNotice(
+			translate( 'Disconecting %(siteName)s', { args: { siteName: site.title } } )
+			, { isPersistent: true, showDismiss: false }
+		);
+
+		disconnectSite( site.ID ).then( () => {
+			// Removing the domain from a domain-only site results
+			// in the site being deleted entirely. We need to call
+			// `receiveDeletedSiteDeprecated` here because the site
+			// exists in `sites-list` as well as the global store.
+			disconnectedSiteDeprecated( site );
+			this.props.setAllSitesSelected();
+			removeInfoNotice( notice.noticeId );
+			showSuccessNotice( translate( 'Disconnected %(siteName)s', { args: { siteName: site.title } } ) );
+			recordGAEvent( 'Jetpack', 'Successfully Disconnected' );
+		}, () => {
+			removeInfoNotice( notice.noticeId );
+			showErrorNotice( translate( 'Error Disconnecting %(siteName)s', { args: { siteName: site.title } } ) );
```

Comment:
Hi! I've found a possible matching string that has already been translated **21** times:
`translate( 'Error disconnecting %(siteName)s.' )` *ES Score: 18*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/disable-jetpack-dialog)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).